### PR TITLE
Feature/shortcuts

### PR DESCRIPTION
**Pull request by YuriyGorvitovskiy**

### CODE REVIEW

File: examples/autocomplete-atomic/index.tsx [hunk 1]
```diff
@@ -52,32 +79,61 @@ const CityChip: FunctionComponent<any> = (props) => {
...
                         triggerChar: "/",
-                        atomicBlockName: "my-city"
+                        atomicBlockName: "my-city",
+                        minSearchChars: 1,
+                    },
+                    {
+                        items: shortcutBullet,
+                        triggerChar: "*",
+                        minSearchChars: 1,
+                        handleAutoComplete: (editorState: EditorState, selectionState: SelectionState, value: string): EditorState => {
```
Comment (agavrilov) [inline examples/autocomplete-atomic/index.tsx]: Extract this function into a standalone function

File: examples/autocomplete-atomic/index.tsx [hunk 2]
```diff
@@ -52,32 +79,61 @@ const CityChip: FunctionComponent<any> = (props) => {
...
+                            editorState = RichUtils.toggleBlockType(editorState, value);
+                            editorState = EditorState.forceSelection(editorState, editorState.getCurrentContent().getSelectionAfter())
+                            return editorState;
+                        }
+                    },
+                    {
+                        items: shortcutNumber,
+                        triggerChar: "1",
+                        minSearchChars: 2,
+                        handleAutoComplete: (editorState: EditorState, selectionState: SelectionState, value: string): EditorState => {
```
Comment (agavrilov) [inline examples/autocomplete-atomic/index.tsx]: Extract this function into a standalone function

File: src/MUIRichTextEditor.tsx [hunk 3]
```diff
@@ -952,9 +1002,14 @@ const MUIRichTextEditor: RefForwardingComponent<TMUIRichTextEditorRef, IMUIRichT
             && keyBinding === "backspace"
             && searchTerm.length) {
             setSearchTerm(searchTerm.substr(0, searchTerm.length - 1))
+            if (autocompletMinSearchCharCount() <= searchTerm.length + 1 && getAutocompleteItems().length > 0) {
```
Comment (agavrilov) [inline src/MUIRichTextEditor.tsx]: The code is repeated twice. Can you extract into a separate function?

File: examples/autocomplete-atomic/index.tsx (hunk 1 above)
Comment (YuriyGorvitovskiy) [reply examples/autocomplete-atomic/index.tsx]: Addressed

File: examples/autocomplete-atomic/index.tsx [hunk 4]
```diff
@@ -42,6 +53,22 @@ const cities: TAutocompleteItem[] = [
         content: "Osaka",
     }
 ]
+const shortcutBullet: TAutocompleteItem[] = [
+    {
+        keys: [""],
```
Comment (agavrilov) [inline examples/autocomplete-atomic/index.tsx:61]: Does it really matter what keys are here? If not, can add a comment about it?

File: examples/autocomplete-atomic/index.tsx [hunk 5]
```diff
@@ -42,6 +53,22 @@ const cities: TAutocompleteItem[] = [
         content: "Osaka",
     }
 ]
+const shortcutBullet: TAutocompleteItem[] = [
```
Comment (agavrilov) [inline examples/autocomplete-atomic/index.tsx:59]: Please, put comments describing what this autocomplete item supposed to do

**Review by agavrilov — CHANGES_REQUESTED** [review]

File: examples/autocomplete-atomic/index.tsx (hunk 4 above)
Comment (YuriyGorvitovskiy) [reply examples/autocomplete-atomic/index.tsx:61]: yes there is filter code that check the keys, and it is important.

File: examples/autocomplete-atomic/index.tsx (hunk 2 above)
Comment (YuriyGorvitovskiy) [reply examples/autocomplete-atomic/index.tsx]: Addressed

File: src/MUIRichTextEditor.tsx (hunk 3 above)
Comment (YuriyGorvitovskiy) [reply src/MUIRichTextEditor.tsx]: Addressed

File: examples/autocomplete-atomic/index.tsx (hunk 5 above)
Comment (YuriyGorvitovskiy) [reply examples/autocomplete-atomic/index.tsx:59]: Addressed

File: examples/autocomplete-atomic/index.tsx [hunk 6]
```diff
@@ -43,6 +54,27 @@ const cities: TAutocompleteItem[] = [
     }
 ]
 
+// This autocomplet will change the block style to unordered-list-item
```
Comment (agavrilov) [inline examples/autocomplete-atomic/index.tsx]: autocomplet**e**

File: examples/autocomplete-atomic/index.tsx [hunk 7]
```diff
@@ -43,6 +54,27 @@ const cities: TAutocompleteItem[] = [
...
+// Autocomplete sequence is * + space
+const shortcutBullet: TAutocompleteItem[] = [
+    {
+        keys: [""],
+        value: "unordered-list-item",
+        content: ""
+    }
+]
+
+// This autocomplet will change the block style to ordered-list-item
```
Comment (agavrilov) [inline examples/autocomplete-atomic/index.tsx]: autocomplet**e**

Comment (YuriyGorvitovskiy) [reply examples/autocomplete-atomic/index.tsx]: Thanks

File: src/MUIRichTextEditor.tsx [hunk 8]
```diff
@@ -477,34 +514,41 @@ const MUIRichTextEditor: RefForwardingComponent<TMUIRichTextEditorRef, IMUIRichT
...
+        if (chars === " " && autocompleteRef.current?.items?.length == 1) {
+            if (0 <= autocompleteRef.current?.items[0].keys.findIndex(k => k === searchTerm)) {
+                handleAutocompleteSelected()
+                return "handled"
+            } else {
+                clearSearch();
+            }
         } else if (autocompleteSelectionStateRef.current) {
             setSearchTerm(searchTerm + chars)
+            if (canShowAutocompletPopup()) {
```
Comment (agavrilov) [inline src/MUIRichTextEditor.tsx]: Autocomplet**e**
```suggestion
            setShowAutocompletePopup(canShowAutocompletePopup());
```

Comment (YuriyGorvitovskiy) [reply src/MUIRichTextEditor.tsx]: Thanks!

**Review by agavrilov — CHANGES_REQUESTED** [review]
Several automplet**e**-s

**Comment by YuriyGorvitovskiy** [comment]
I think I resolve all of them with last commit

**Review by agavrilov — APPROVED** [review]

**Pull request merged**